src/core/qgsexpressioncontext.cpp

```diff
@@ -630,21 +630,12 @@ QgsExpressionContextScope* QgsExpressionContextUtils::projectScope()
 
   QgsExpressionContextScope* scope = new QgsExpressionContextScope( QObject::tr( "Project" ) );
 
-  //add variables defined in project file
-  QStringList variableNames = project->readListEntry( "Variables", "/variableNames" );
-  QStringList variableValues = project->readListEntry( "Variables", "/variableValues" );
+  QgsStringMap vars = QgsProject::instance()->variables();
 
-  int varIndex = 0;
-  Q_FOREACH ( const QString& variableName, variableNames )
```

Comment:
Since we are using C++11 now... there is also support for for-based range loops over maps. But the std::map one looks a lot nicer. I wonder if we should consider moving at some point ;)

    const auto vars = QgsProject::instance()->variables().toStdMap();
    for ( const auto& pair : vars )
    {
      scope->setVariable( pair.first, pair.second );
    }

vs

    const QgsStringMap vars = QgsProject::instance()->variables();
    for ( const auto& key : vars )
    {
      scope->setVariable( key, vars.value( key ) );
    }